modules/navier_stokes/src/fv/kernels/INSFVMomentumAdvection.C

```diff
@@ -191,25 +227,62 @@ INSFVMomentumAdvection::coeffCalculator(const Elem & elem, const ADReal & mu) co
                                                             const bool /*elem_has_info*/) {
     mooseAssert(fi, "We need a non-null FaceInfo");
 
-    const Point normal = (neighbor == fi->neighborPtr()) ? fi->normal() : Point(-fi->normal());
+    const bool rc_elem_is_fi_elem = (neighbor == fi->neighborPtr());
+    const Point normal = rc_elem_is_fi_elem ? fi->normal() : Point(-fi->normal());
+
+    // Unless specified otherwise, "elem" here refers to the element we're computing the
+    // Rhie-Chow coefficeint for. "neighbor" is the element across the current FaceInfo (fi)
+    // face from the Rhie-Chow element
 
     if (fi->isBoundary())
     {
-      bool bc_found = false;
-
       // In my mind there should only be about one bc_id per FaceInfo
       for (const auto bc_id : fi->boundaryIDs())
+      {
         if (_no_slip_wall_boundaries.find(bc_id) != _no_slip_wall_boundaries.end())
         {
+          // Need to account for viscous shear stress from wall
           for (const auto i : make_range(_dim))
             coeff(i) += mu * surface_vector.norm() /
                         std::abs((fi->faceCentroid() - fi->elemCentroid()) * normal) *
                         (1 - normal(i) * normal(i));
-          bc_found = true;
+
+          // No flow normal to wall, so no contribution to coefficient from the advection term
+          break;
+        }
+
+        if (_flow_boundaries.find(bc_id) != _flow_boundaries.end())
+        {
+          ADRealVectorValue face_velocity(_u_var->getBoundaryFaceValue(*fi));
+          if (_v_var)
+            face_velocity(1) = _v_var->getBoundaryFaceValue(*fi);
+          if (_w_var)
+            face_velocity(2) = _w_var->getBoundaryFaceValue(*fi);
+
+          const auto advection_coeffs = Moose::FV::interpCoeffs(
+              _advected_interp_method, *fi, rc_elem_is_fi_elem, face_velocity);
+          ADReal temp_coeff = _rho * face_velocity * surface_vector * advection_coeffs.first;
+
+          // For flow boundaries, the coefficient addition is the same for every velocity component
+          for (const auto i : make_range(_dim))
+            coeff(i) += temp_coeff;
+
+          // No wall exerting shear on the fluid, so no contribution to the coefficient from the
+          // viscous term
           break;
         }
 
-      // if (!bc_found)
+        if (_slip_wall_boundaries.find(bc_id) == _slip_wall_boundaries.end())
+          mooseError(
+              "If INSFVMomentumAdvection is to be executed along a boundary, then that boundary "
+              "must be included in one of the following parameters: 'no_slip_wall_boundaries', "
+              "'slip_wall_boundaries', or 'flow_boundaries'. However, the current boundary ID ",
+              bc_id,
+              " is not currently covered.");
+
+        // In the case of a slip wall we neither have viscous shear stress from the wall nor normal
+        // outflow, so our contribution to the coefficient is zero
```

Comment:
I really like all these comments! Super helpful to understand the physics in the code.